typescript/cli/src/validator/preFlightCheck.ts

```diff
@@ -0,0 +1,117 @@
+import { MerkleTreeHook__factory } from '@hyperlane-xyz/core';
+import { HyperlaneCore, S3Validator } from '@hyperlane-xyz/sdk';
+import { Address } from '@hyperlane-xyz/utils';
+
+import { CommandContext } from '../context/types.js';
+import { errorRed, logBlue, logGreen, warnYellow } from '../logger.js';
+
+export const checkValidatorSetup = async (
```

Comment:
similar to what i said in the other PR, maybe useful to split up chunks in here to other functions just to help readability with all the different branching, but don't feel obligated to

Reply:
I will be pulling out some of this logic into functions to be reused by the AVS validator check command